src/drivers/cmakeDriver.ts

```diff
@@ -1730,6 +1747,67 @@ export abstract class CMakeDriver implements vscode.Disposable {
         }
     }
 
+    private async _showTaskQuickPick(tasks: vscode.Task[]): Promise<vscode.Task | undefined> {
+        interface TaskItem extends vscode.QuickPickItem {
+            task: vscode.Task;
+        }
+        const choices = tasks.map((t): TaskItem => ({ label: t.name, task: t }));
+        const sel = await vscode.window.showQuickPick(choices, { placeHolder: localize('select.build.task', 'Select build task') });
+        return sel ? sel.task : undefined;
+    }
+
+    private async _findBuildTask(targets?: string[]): Promise<vscode.Task | undefined> {
+
+        // Fetch all CMake task
+        const tasks = (await vscode.tasks.fetchTasks({ type: 'cmake' })).filter(task => ((task.group?.id === vscode.TaskGroup.Build.id)));
+
+        // There is only one found - stop searching
+        if (tasks.length === 1) {
+            return tasks[0];
```

Comment:
I didn't think the extension currently picked any task that wasn't marked as default.  This is a change in policy.

Reply:
AFAIK it does.
Extension currently adds one task of each group (build, clean, install...).
These arent markes as 'default' and for example build task was used in case that no other tasks were present in tasks.json file.